### DISCUSSION TITLE
Chaging passwd while passfile is used

### DISCUSSION THREAD

**Discussion by KrishCrS** [rfjakob/gocryptfs#882] [Q&A] [answered]
```
gocryptfs -passfile password.txt -passwd -d CIPHER
```
```
OpenSSL disabled, using Go GCM
passfile: reading from file "password.txt"
Decrypting master key
CryptoCore.Wipe: Only nil'ing stdlib refs
Please enter your new password.
passfile: reading from file "password.txt"
CryptoCore.Wipe: Only nil'ing stdlib refs
Password changed.
```

I generated a binary file with the name `passwod.txt`, using `/dev/urandom`, so I can't pass the password from STDIN. Can I change the password in this scenario?

**Comment by rfjakob** [comment]
Interesting situation :)

How about this (make a copy of your `gocryptfs.conf` beforehand!):

```
echo "" > newline
echo "the new password 123324234!" > newpassword
cat password.txt newline newpassword | gocryptfs -passwd
```

**Reply by rfjakob** [reply, marked as answer]
Alternatively, download the latest source code and compile, I now have `-passwd` ignore `-passfile` ( https://github.com/rfjakob/gocryptfs/commit/d5bd98eb3f4cbfb8dd9d0b2eb64dbff69c3c88b1 )
